frontends/PyCDE/src/pycde/esi.py

```diff
@@ -624,3 +627,120 @@ def package(sys: System):
     circt_lib_dir = build_dir / "tools" / "circt" / "lib"
     esi_lib_dir = circt_lib_dir / "Dialect" / "ESI"
   # shutil.copy(esi_lib_dir / "ESIPrimitives.sv", sys.hw_output_dir)
+
+
+def ChannelDemux2(data_type: Type):
+
+  class ChannelDemux2(Module):
+    """Combinational 2-way channel demultiplexer for valid/ready signaling."""
+
```

Comment:
For any practical application, won't this completely kill fmax (under the assumption of a sufficiently large demux tree). I'm in favor of doing this for the initial POC, but this'll obviously have to change in the future...

Reply:
The thought is to add buffering at a higher level in the instance hierarchy. (While assembling the tree.)

Reply:
I plan on adding buffers on the input and output of the Channel[De]mux function after the next PR.